pkg/csi/service/wcp/controller.go

```diff
@@ -1278,28 +1279,24 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 						return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
 							"failed to connect to Virtual Center: %s", vc.Config.Host)
 					}
-					podVM, err := getVMByInstanceUUIDInDatacenter(ctx, vc, dcMorefValue, v)
-					if err != nil {
-						if err == cnsvsphere.ErrVMNotFound {
-							log.Infof("virtual machine not found for vmUUID %q. "+
-								"Thus, assuming the volume is detached.", v)
-							break
-						}
-						return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
-							"failed to get the PodVM Moref from the PodVM UUID: %s in datacenter: %s with err: %+v",
-							v, dcMorefValue, err)
-					}
 					isStillAttached := false
 					timeout := 4 * time.Minute
 					pollTime := time.Duration(5) * time.Second
-					err = wait.Poll(pollTime, timeout, func() (bool, error) {
-						podVM, err := getVMByInstanceUUIDInDatacenter(ctx, vc, dcMorefValue, v)
+					var podVM *cnsvsphere.VirtualMachine
+					err = wait.PollImmediate(pollTime, timeout, func() (bool, error) {
+						podVM, err = getVMByInstanceUUIDInDatacenter(ctx, vc, dcMorefValue, v)
 						if err != nil {
 							if err == cnsvsphere.ErrVMNotFound {
 								log.Infof("virtual machine not found for vmUUID %q. "+
 									"Thus, assuming the volume is detached.", v)
 								return true, err
 							}
+							if err == cnsvsphere.ErrInvalidVC {
+								log.Errorf("failed to get the PodVM Moref from the PodVM UUID: %s in datacenter: "+
+									"%s with err: %+v", v, dcMorefValue, err)
+								return true, err
+							}
+							return false, nil
```

Comment:
Shouldn't it be `return false, err`?

Reply:
if err is returned other than `ErrVMNotFound` and `ErrInvalidVC`, we want to continue the loop.